src/aiotaskq/worker.py

```diff
@@ -248,31 +240,64 @@ async def _execute_task_and_publish(
         self,
         pubsub: IPubSub,
         task: "Task",
-        task_args: list,
-        task_kwargs: dict,
-        task_id: str,
         semaphore: t.Optional["asyncio.Semaphore"],
     ):
         self._logger.debug(
             "[%s] Executing task %s(*%s, **%s)",
-            *(self._pid, task_id, task_args, task_kwargs),
+            *(self._pid, task.id, task.args, task.kwargs),
         )
-        if inspect.iscoroutinefunction(task.func):
-            task_result = await task(*task_args, **task_kwargs)
-        else:
-            task_result = task(*task_args, **task_kwargs)
 
-        # Publish the task return value
-        self._logger.debug(
-            "[%s] Publishing task result %s(*%s, **%s)",
-            *(self._pid, task_id, task_args, task_kwargs),
-        )
-        task_result = json.dumps(task_result)
-        result_channel = RESULTS_CHANNEL_TEMPLATE.format(task_id=task_id)
-        await pubsub.publish(channel=result_channel, message=task_result)
+        retry = False
+        error = None
+        retries: int = None
+        retry_max: int | None = None
+        task_result: t.Any = None
+        try:
+            if inspect.iscoroutinefunction(task.func):
+                task_result = await task(*task.args, **task.kwargs)
+            else:
+                task_result = task(*task.args, **task.kwargs)
+        except Exception as e:  # pylint: disable=broad-except
+            error = e
+            if task.retry is not None:
+                retry_max = task.retry["max_retries"]
+                if isinstance(e, task.retry["on"]):
+                    retry = True
+
+        finally:
+            # Retry if still within retry limit
+            if retry:
+                async with redis.from_url(url="redis://localhost:6379") as redis_client:
+                    retries = int(await redis_client.get(f"retry:{task.id}") or 0)
```

Comment:
If `redis_client.get` keeps returning `None` or something, it might mean we will never stop retrying

Reply:
Good catch